skrub/datasets/_fetching.py

```diff
@@ -718,6 +724,13 @@ def fetch_employee_salaries(
             )
         if drop_irrelevant:
             dataset.X.drop(["full_name"], axis=1, inplace=True)
+        if overload_job_titles:
```

Comment:
I agree it would make the code slightly more simple, but at the cost of aliasing `X` beforehand, and overwriting it in `dataset` afterwards (since we don't necessarily modify it in-place). I don't think this is very important. Wdyt @jovan-stojanovic?

Reply:
Hey, IMO let's go for the most efficient, both are readable enough. I think it's important to be as fast as possible when importing data in examples.

Reply:
Well the cost I talked about was on the number of lines and error-proneness, not so much on the actual computation cost :)
I guess both versions are equal in that sense, thus it's purely a question of readability.

Reply:
I guess the answer is as you wish then :laughing:

Reply:
Well, let's keep it like that for now then, it'll probably change with #669 anyway :)